protobuf/src/main/java/io/grpc/protobuf/StatusProto.java

```diff
@@ -164,6 +165,15 @@ private static Metadata toMetadata(com.google.rpc.Status statusProto, Metadata m
         return statusProto;
       }
     }
+    // fall-back to status, this is useful if the error is local. e.g. Server is unavailable.
+    if (status != null) {
```

Comment:
Optional: It seems like `status` should not be null,  might be cleaner to use a precondition rather than having to put an if-block here. Although we already do a check for null trailers...should the `Metadata` param have a `@Nullable` annotation? Looking [here](https://github.com/grpc/grpc-java/blob/master/api/src/main/java/io/grpc/StatusException.java#L51) is seems like the general "convention" is that status is non-null but trailers can be null, although that doesn't have to match the convention here.

Reply:
I didn't do it because it is api change, but since it is experimental and you have same impression, might as well fix it. also removed `@Nullable` from return type.